core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -659,6 +659,9 @@ trait WorkspaceComponent {
 
     def setIsLocked(isLocked: Boolean): WriteAction[Boolean] =
       query.map(_.isLocked).filter(_ =!= isLocked).update(isLocked).map(_ > 0)
+
+    def setState(state: WorkspaceState): WriteAction[Int] =
```

Comment:
@trholdridge there's already an `updateState` method up at line 404. Did you try using that?

Reply:
I created the new one because the other DB updates in that part of `BillingAccountChangeSynchronizer` extend an existing query. I wasn't sure how to back the `workspaceId` out of that query to pass to the existing `updateState` function. I will look into that though. Would be good to rule it out as a source of error.